### PR TITLE
fix(module:upload): fix upload drag drop will open new tab in firefox 91 and 92

### PR DESCRIPTION
**Pull request by small-lady**
…91 and 92

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
解决nz-upload在火狐浏览器91及92版本上拖拽上传时会默认打开一个新窗口

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: components/upload/upload.component.ts [hunk 1]
```diff
@@ -344,12 +351,17 @@ export class NzUploadComponent implements OnInit, OnChanges, OnDestroy {
     });
   }
 
+  ngAfterViewInit(): void {
+    document.body.addEventListener('drop', this.handleDropOpenNewTab);
+  }
```
Comment (arturovt) [inline components/upload/upload.component.ts:356]:
```suggestion
  constructor(private ngZone: NgZone, @Inject(DOCUMENT) private document: Document) {}

  ngAfterViewInit(): void {
    this.ngZone.runOutsideAngular(() => fromEvent(this.document.body, 'drop').pipe(takeUntil(this.destroy$)).subscribe(event => {
      event.preventDefault();
      event.stopPropagation();
    }));
  }
```

File: components/upload/upload.component.ts [hunk 2]
```diff
@@ -344,12 +351,17 @@ export class NzUploadComponent implements OnInit, OnChanges, OnDestroy {
...
+
   ngOnChanges(): void {
     this.zipOptions().setClassMap();
   }
 
   ngOnDestroy(): void {
     this.destroy$.next();
     this.destroy$.complete();
+    document.body.removeEventListener('drop', this.handleDropOpenNewTab)
```
Comment (arturovt) [inline components/upload/upload.component.ts:365]:
```suggestion

```

File: components/upload/upload.component.ts [hunk 3]
```diff
@@ -328,6 +329,12 @@ export class NzUploadComponent implements OnInit, OnChanges, OnDestroy {
     this.cdr.detectChanges();
   }
 
+  // fix firefox drop open new tab
+  private handleDropOpenNewTab(e: DragEvent){
+    e.preventDefault();
+    e.stopPropagation()
+  }
+
```
Comment (arturovt) [inline components/upload/upload.component.ts:337]:
```suggestion
```

**Review by arturovt — CHANGES_REQUESTED** [review]

**Comment by cipchk** [comment]
@small-lady Record a screen? I don't see this prompt here. My firefox env: Mac 95.0.2 (64-bit).

**Comment by small-lady** [comment]
> @small-lady Record a screen? I don't see this prompt here. My firefox env: Mac 95.0.2 (64-bit).

This problem is not in 95.0.2

**Comment by simplejason** [comment]
> > @small-lady Record a screen? I don't see this prompt here. My firefox env: Mac 95.0.2 (64-bit).
> 
> This problem is not in 95.0.2 @cipchk ； it is in 91 and 92(Firefox Extended Support Release)

@cipchk Any question about this PR?

File: components/upload/upload.component.ts [hunk 4]
```diff
@@ -175,6 +179,8 @@ export class NzUploadComponent implements OnInit, OnChanges, OnDestroy {
   // #endregion
 
   constructor(
+    private ngZone: NgZone,
+    @Inject(DOCUMENT) private document: Document,
```
Comment (cipchk) [inline components/upload/upload.component.ts:183]:
```suggestion
    @Inject(DOCUMENT) private document: NzSafeAny,
```

建议使用 `NzSafeAny` 来代替，不然可能在 SSR 下会有问题。

**Review by cipchk — CHANGES_REQUESTED** [review]

File: components/upload/upload.component.ts (hunk 4 above)
Comment (small-lady) [reply components/upload/upload.component.ts]: @cipchk 已按建议修改，谢谢~

**Comment by cipchk** [comment]
LGTM

**Pull request merged**